CommonTools/Utils/interface/KinematicColumns.h

```diff
@@ -9,11 +9,25 @@ namespace edm {
 
     namespace col {
 
-      SOA_DECLARE_COLUMN(Pt, float, "pt");
       SOA_DECLARE_COLUMN(Eta, float, "eta");
       SOA_DECLARE_COLUMN(Phi, float, "phi");
+      SOA_DECLARE_COLUMN(Theta, float, "theta");
+
       SOA_DECLARE_COLUMN(Vz, float, "vz");
 
+      SOA_DECLARE_COLUMN(Px, float, "px");
+      SOA_DECLARE_COLUMN(Py, float, "py");
+      SOA_DECLARE_COLUMN(Pz, float, "pz");
+      SOA_DECLARE_COLUMN(Pt, float, "pt");
+      SOA_DECLARE_COLUMN(P, float, "p");
+
+      SOA_DECLARE_COLUMN(PtError, float, "ptError");
+      SOA_DECLARE_COLUMN(D0, float, "d0");
+
+      SOA_DECLARE_COLUMN(NumberOfValidHits, int, "numberOfValidHits");
+      SOA_DECLARE_COLUMN(MissingInnerHits, int, "missingInnerHits");
+      SOA_DECLARE_COLUMN(Charge, int, "charge");
```

Comment:
these are not kinematic variables, they do not belong in KinematicColumns.h
Smth like TrackSpecificColumns or similar would be the place following the structure proposed earlier.

Also, I thought that the SoA will ideally be needed just for the heavy methods and will not need to cover inlined methods like px,py,pz, d0, charge.
Otherwise, this is becoming a TrackSoA of sorts

Reply:
Yes I'm aware, these columns are just declared here for an initial solution because I didn't know where it should actually go (see https://github.com/cms-sw/cmssw/pull/30956#discussion_r462239779).

Good point with the heavy methods. Indeed, my reasoning to have these inline columns in the SOA as well was not performance, but consistency. If we kept this information out of the SOA, the `findConversion` function (https://github.com/cms-sw/cmssw/pull/30956/files#diff-b2e29989d42a554a669a74669f9fb7e0R61) would have to take again (references to) both track collections as parameters and take care to align the iteration over the SOAs and the track collections.

Also, remember that SOA makes also sense for inline functions for potential vectorization.

Actually, `px`, `py`, `pz`, `vz` and `p` are not used by the conversion finder after all. It was in there before because some LorentzVectors were created, but I realized later they were unnecessary. That leaves us with the track specific columns `d0`, `numberOfValidHits`, `missingInnerHits` and `charge`. Would you introduce `TrackSpecificColumns.h` just for them? We could either do that, or alternatively define them in the conversion finder for local use.

Reply:
If the full Track is desired to be as a SoA, then TrackSoA or TrackTable makes sense, but then it sounds a bit too generic and will likely cost more to create and fill (unless it's a lazy fill by column).

If the goal is to really represent the full track in a generic way, then moving this to DataFormats/TrackReco apparently makes sense, but currently it seems like an overkill for the issue at hand (eta, phi, and perhaps "nHits" costs)

Reply:
If the full track will be floating in memory as an SoA, I'd rather just add eta, phi and alike as transients in the Track class (so that track.phi() is ```return phi_```) to get a very likely cheaper solution.

Reply:
Hi Slava, while I agree that storing the expensive values directly in the track to slve the problem of expensive math functions in a more sustainable way, I don't think it's a "cheaper solution" at this point:

* changes in the Track DataFormat go beyond the goal of this PR to speed up the conversion finding for electrons
* SOA is probably still better due to cache locality and vectorization possibilities
* I don't have the time right now to change everything again and redo the profiling

So I would kindly ask you to consider the SOA solution as proposed in this PR for the speedup envisioned in issue https://github.com/cms-sw/cmssw/issues/30954.